src/utils/Deprecation.ts

```diff
@@ -0,0 +1,25 @@
+export class Deprecation {
+
+  private _deprecationWarning: boolean;
+
+  constructor (deprecationWarning: boolean) {
+    this._deprecationWarning = deprecationWarning;
+  }
+
+  /**
+   * Warn the developer that he is using a deprecated action
+   * 
+   * @param response Result of a query to the API
+   * 
+   * @returns Same as response param, just like a middleware
+   */
+  logDeprecation (response) {
+    if (this._deprecationWarning && response.deprecations && response.deprecations.length) {
+      for (let index = 0; index < response.deprecations.length; index++) {
```

Comment:
Prefer `for (const ... of)` loop:
 - readability: no need to access elements by the array index
 - security: no out of range errors